nixos/modules/services/x11/desktop-managers/elementary.nix

```diff
@@ -0,0 +1,230 @@
+        default = with pkgs; [
+          elementary.gala
+          epiphany
+          elementary.elementary-settings-daemon
+          gnome3.mutter
+          gtk3
+          plank
+        ];
+        type = types.listOf types.path;
+        description = "List of packages for which gsettings are overridden.";
+      };
+
+      debug = mkEnableOption "gnome-session debug messages";
+
+    };
+
+    environment.elementary.excludePackages = mkOption {
+      default = [];
+      example = literalExample "[ pkgs.elementary.elementary-camera ]";
+      type = types.listOf types.package;
+      description = "Which packages elementary should exclude from the default environment";
+    };
+
+  };
+
+
+  config = mkIf cfg.enable {
+
+    services.xserver.displayManager.extraSessionFilePackages = [ pkgs.elementary.elementary-session-settings ];
+
+    services.xserver.displayManager.sessionCommands = ''
+      if test "$XDG_CURRENT_DESKTOP" = "Pantheon"; then
+          ${concatMapStrings (p: ''
+            if [ -d "${p}/share/gsettings-schemas/${p.name}" ]; then
+              export XDG_DATA_DIRS=$XDG_DATA_DIRS''${XDG_DATA_DIRS:+:}${p}/share/gsettings-schemas/${p.name}
+            fi
+
+            if [ -d "${p}/lib/girepository-1.0" ]; then
+              export GI_TYPELIB_PATH=$GI_TYPELIB_PATH''${GI_TYPELIB_PATH:+:}${p}/lib/girepository-1.0
+              export LD_LIBRARY_PATH=$LD_LIBRARY_PATH''${LD_LIBRARY_PATH:+:}${p}/lib
+            fi
+          '') cfg.sessionPath}
+
+          # Horrible aweful workaround for https://github.com/NixOS/nixpkgs/issues/45043
+          export XDG_DATA_DIRS=$XDG_DATA_DIRS''${XDG_DATA_DIRS:+:}${pkgs.networkmanagerapplet}/share
+
+          # Makes qt applications look less alien
+          export QT_QPA_PLATFORMTHEME=gtk3
+          export QT_STYLE_OVERRIDE=adwaita
+      fi
+    '';
+
+    # Override GSettings schemas
+    environment.variables.NIX_GSETTINGS_OVERRIDES_DIR = "${nixos-gsettings-desktop-schemas}/share/gsettings-schemas/nixos-gsettings-overrides/glib-2.0/schemas";
+
+    # So gnome-session can find the pantheon session
+    # This also sets the default mimeapps
+    environment.variables.XDG_DATA_DIRS = [ "${pkgs.elementary.elementary-session-settings}/share" ];
+
+    environment.variables.GNOME_SESSION_DEBUG = optionalString cfg.debug "1";
+
+    environment.systemPackages = pkgs.elementary.artwork ++ pkgs.elementary.desktop ++ pkgs.elementary.services
+      ++ (removePackagesByName pkgs.elementary.apps config.environment.elementary.excludePackages)
+      ++ (with pkgs.gnome3;
+      [
+        adwaita-icon-theme
+        dconf
+        epiphany
+        evince
+        file-roller
+        geary
+        gnome-bluetooth
+        gnome-font-viewer
+        gnome-menus
+        gnome-power-manager
+      ])
+      ++ (with pkgs;
+      [
+        adwaita-qt
+        desktop-file-utils
+        glib
+        glib-networking
+        gtk3.out
+        gvfs
```

Comment:
I would guess this collides with the `gvfs` added by the `services.gnome3.gvfs.enable` option.

Reply:
Yep that sounds right.